homeassistant/components/mawaqit/mawaqit_wrapper.py

```diff
@@ -0,0 +1,184 @@
+"""Provides a wrapper for interacting with the MAWAQIT API.
+
+It includes functions for testing credentials, retrieving API tokens,
+fetching prayer times, and finding mosques in the neighborhood.
+"""
+
+import logging
+
+from mawaqit import AsyncMawaqitClient
+from mawaqit.consts import BadCredentialsException
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def validate_credentials(
+    username: str | None = None,
+    password: str | None = None,
+    client_instance: AsyncMawaqitClient | None = None,
+) -> bool:
+    """Return True if the MAWAQIT credentials is valid."""
+    try:
+        client = client_instance
+        if client is None:
+            client = AsyncMawaqitClient(username=username, password=password)
+        await client.login()
+    except BadCredentialsException:
+        _LOGGER.error("Error : Bad Credentials")
```

Comment:
There's no reason to log these kind of expected errors with severity `error`, instead, you should tell the user via the config flow what went wrong.

Same comment for the other functions called by the config flow.